File: android-app/app/src/main/java/com/example/chatrt/models/Reminder.java
package com.example.chatrt.models;

import com.google.gson.annotations.SerializedName;
import java.io.Serializable;
import java.util.Date;

public class Reminder implements Serializable {
    @SerializedName("_id")
    private String id;
    private String conversationId;
    private User creatorId;
    private User partnerId;
    private String content;
    private Date dueDate;
    private Date createdAt;

    public String getId() { return id; }
    public String getConversationId() { return conversationId; }
    public User getCreatorId() { return creatorId; }
    public User getPartnerId() { return partnerId; }
    public String getContent() { return content; }
    public Date getDueDate() { return dueDate; }
    public Date getCreatedAt() { return createdAt; }
}

File: BACK-END/src/models/Reminder.js
import mongoose from "mongoose";

const reminderSchema = new mongoose.Schema({
  conversationId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "Conversation",
    required: true,
  },
  creatorId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "USERS",
    required: true,
  },
  partnerId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "USERS",
    required: true,
  }, // Người cùng hội thoại
  content: { type: String, required: true },
  dueDate: { type: Date, required: true },
  createdAt: { type: Date, default: Date.now },
});

const Reminder = mongoose.model("Reminder", reminderSchema);
export default Reminder;
